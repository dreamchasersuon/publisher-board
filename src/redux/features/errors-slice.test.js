import configureStore from 'redux-mock-store';
import errors, { setError } from './errorsFeatureSlice';

describe('redux set and remove errors', () => {
  test('action with error', () => {
    const mockStore = configureStore();
    const store = mockStore({});
    store.dispatch(
      setError({
        e: { http_status_code: '422', message: 'Invalid request data' },
      }),
    );
    expect(store.getActions()).toMatchSnapshot();
  });

  test('reducer should handle setError action', () => {
    expect(
      errors([], {
        type: 'errors/setError',
        payload: {
          e: { http_status_code: '400', message: 'Network error' },
        },
      }),
    ).toMatchSnapshot();
  });

  test('reducer should handle removeError action', () => {
    expect(
      errors([{ http_status_code: '400', message: 'Network error' }], {
        type: 'errors/removeError',
      }),
    ).toMatchSnapshot();
  });
});
