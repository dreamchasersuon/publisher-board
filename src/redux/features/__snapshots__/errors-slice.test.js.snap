// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`redux set and remove errors action with error 1`] = `
Array [
  Object {
    "payload": Object {
      "e": Object {
        "http_status_code": "422",
        "message": "Invalid request data",
      },
    },
    "type": "errors/setError",
  },
]
`;

exports[`redux set and remove errors reducer should handle removeError action 1`] = `Array []`;

exports[`redux set and remove errors reducer should handle setError action 1`] = `
Array [
  Object {
    "http_status_code": "400",
    "message": "Network error",
  },
]
`;
